src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs

```diff
@@ -1454,7 +1454,7 @@ public static FormulaValue Table(IRContext irContext, FormulaValue[] args)
                     BlankValue b => DValue<RecordValue>.Of(b),
                     _ => DValue<RecordValue>.Of((ErrorValue)arg),
                 });
-            return new InMemoryTableValue(irContext, records);
+            return new InMemoryTableValue(irContext, records.ToList());
```

Comment:
>.ToList() [](http://example.com/codeflow?start=60&length=9)

We should add a comment here .... this ToList() is ensuring that the returned table is mutable.